fdb-record-layer-lucene/src/test/java/com/apple/foundationdb/record/lucene/LuceneIndexTest.java

```diff
@@ -0,0 +1,359 @@
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+
+/**
+ * Tests for {@code FULL_TEXT} type indexes.
+ */
+@Tag(Tags.RequiresFDB)
+public class LuceneIndexTest extends FDBRecordStoreTestBase {
+
+    private static final Index SIMPLE_TEXT_SUFFIXES = new Index("Simple$text_suffixes", field("text"), LuceneIndexTypes.LUCENE,
+            ImmutableMap.of(IndexOptions.TEXT_TOKENIZER_NAME_OPTION, AllSuffixesTextTokenizer.NAME));
+
+    private static final Index COMPLEX_MULTIPLE_TEXT_INDEXES = new Index("Complex$text_multipleIndexes", concatenateFields("text", "text2"), LuceneIndexTypes.LUCENE,
+            ImmutableMap.of(IndexOptions.TEXT_TOKENIZER_NAME_OPTION, AllSuffixesTextTokenizer.NAME));
+
+    private static final String DYLAN = "You're an idiot, babe\n" +
+                                        "It's a wonder that you still know how to breathe";
+
+    private static final String WAYLON = "There's always one more way to do things and that's your way, and you have a right to try it at least once.";
+
+    protected void openRecordStore(FDBRecordContext context, FDBRecordStoreTestBase.RecordMetaDataHook hook) {
+        RecordMetaDataBuilder metaDataBuilder = RecordMetaData.newBuilder().setRecords(TestRecordsTextProto.getDescriptor());
+        metaDataBuilder.getRecordType(COMPLEX_DOC).setPrimaryKey(concatenateFields("group", "doc_id"));
+        hook.apply(metaDataBuilder);
+        recordStore = getStoreBuilder(context, metaDataBuilder.getRecordMetaData())
+                .setSerializer(TextIndexTestUtils.COMPRESSING_SERIALIZER)
+                .uncheckedOpen();
+        setupPlanner(null);
+    }
+
+    @Nonnull
+    protected FDBRecordStore.Builder getStoreBuilder(@Nonnull FDBRecordContext context, @Nonnull RecordMetaData metaData) {
+        return FDBRecordStore.newBuilder()
+                .setFormatVersion(FDBRecordStore.MAX_SUPPORTED_FORMAT_VERSION) // set to max to test newest features (unsafe for real deployments)
+                .setKeySpacePath(path)
+                .setContext(context)
+                .setMetaDataProvider(metaData);
+    }
+
+    private TestRecordsTextProto.SimpleDocument createSimpleDocument(long docId, String text, int group) {
+        return TestRecordsTextProto.SimpleDocument.newBuilder()
+                .setDocId(docId)
+                .setText(text)
+                .setGroup(group)
+                .build();
+    }
+
+    private TestRecordsTextProto.ComplexDocument createComplexDocument(long docId, String text, String text2, int group) {
+        return TestRecordsTextProto.ComplexDocument.newBuilder()
+                .setDocId(docId)
+                .setText(text)
+                .setText2(text2)
+                .setGroup(group)
+                .build();
+    }
+
+
+    @Test
+    public void simpleInsertAndSearch() {
```

Comment:
These seem like a good start to the tests, but there are a few things I can think of to add (either now or later), mostly inspired by the bugs I think I saw in the code above:
1. Multiple indexes on the same store
2. Multiple stores with the same index in the same context
3. Complex queries that match a document on both fields, and use an OR
4. calling `scanIndexRecords` and getting the actual document back
5. saving some records, committing, querying and mapping that into an update
6. Adding to at least some of these tests assertions about the resulting index entries.
7. Updating the metadata to cause the index subpsace to change mid-transaction


(Note: I would not be surprised if that list of tests goes noticeably beyond the testing of many of the existing cursors/indexes, but in that case it's an issue there, not here.